scripts/newmoose.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+function printusage {
+    echo "Usage:    newmoose.sh <type> <AppName>"
+    echo ""
+    echo "    Creates a new blank MOOSE app in the current working directory."
+    echo "    <type> must be either "app" or "module"."
+    echo "    <AppName> should be given in CamelCase format."
+}
+
+if [[ "$1" == "-h" || $# != 2 ]]; then
+    printusage
+    exit 1
+fi
+
+if [[ "$1" != "app" && "$1" != "module" ]]; then
+    echo "error: invalid type given" >&2
+    printusage
+    exit 1
+fi
+
+# set old/new app name variables
+srcname='Stork'
+dstname=$2
+kind=$1
+srcnamelow=$(echo "$srcname" | awk '{print tolower($0)}')
+srcnameup=$(echo "$srcname" | awk '{print toupper($0)}')
+dstnamelow=$(echo "$dstname" | awk '{print tolower($0)}')
+dstnameup=$(echo "$dstname" | awk '{print toupper($0)}')
+dir="$dstnamelow"
+if [[ -z $MOOSE_DIR ]]; then
+    MOOSE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/.."
+fi
+if [[ "$kind" == "module" ]]; then
+    dir="${MOOSE_DIR}/modules/$dir"
+fi
+
+absdir=$(echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")")
+if [[ "$kind" == "app" && "$absdir" =~ "$MOOSE_DIR" ]]; then
+    echo "error: your current working directory is inside the MOOSE directory" >&2
+    exit 1
+fi
+
+# make new app dir and copy stork files
+if [[ -d "$dir" ]]; then
+    echo "error: directory '$dir' already exists" >&2
+    exit 1
+fi
+cp -R "$MOOSE_DIR/stork" "$dir"
+find $dir | grep '/[.]' | xargs rm -rf # remove hidden files (e.g. vim swp files)
```

Comment:
Good god this makes me nervous.  Can we put some extra checking on here that `$dir` is NOT their home directory or something?  I don't know how it would get in that state... but we need to be extra careful about these kinds of things.